src/components/molecules/EarthEditor/DatasetPane/DatasetModal/Gdrive/index.tsx

```diff
@@ -0,0 +1,145 @@
+      {pickedFile?.id && (
+        <>
+          <Flex justify="center" align="center">
+            <GdriveIcon size={32} icon="googleDrive" />
+            <Text size="m" color={theme.main.strongText} weight="bold">
+              {intl.formatMessage({ defaultMessage: "Connect with Google Drive" })}
+            </Text>
+          </Flex>
+          <Divider margin="24px" />
+          <Flex justify="center">
+            <AssetCard
+              isImage={false}
+              name={pickedFile.name}
+              cardSize="medium"
+              icon="sheetFile"
+              iconSize="50px"
+              onCheck={() => handleAuthClick()}
+            />
+          </Flex>
+          <Divider margin="24px" />
+          <AssetWrapper direction="column" justify="space-between">
+            <AssetList wrap="nowrap" justify="space-between">
+              {pickedFileSheets?.map((sheetItem: GoolgeSheet) => (
+                <AssetListItem
+                  key={sheetItem.properties.sheetId}
+                  asset={{ id: sheetItem.properties.sheetId, name: sheetItem.properties.title }}
+                  isImage={false}
+                  onCheck={() => {
+                    handleSheetSelect({
+                      id: sheetItem.properties.sheetId,
+                      name: sheetItem.properties.title,
+                    });
+                  }}
+                  selected={sheetItem.properties.sheetId === selectedSheet?.id}
+                />
+              ))}
+            </AssetList>
+          </AssetWrapper>
+          <Divider margin="24px" />
+        </>
+      )}
+
+      {!pickedFile?.id && (
+        <>
+          <Flex justify="center" align="center">
+            <GdriveIcon size={32} icon="googleDrive" />
+            <Text size="m" color={theme.main.strongText} weight="bold">
+              {intl.formatMessage({ defaultMessage: "Connect with Google Drive" })}
+            </Text>
+          </Flex>
+          <Divider margin="24px" />
+          <Flex justify="center">
+            <Text
+              size="m"
+              color={theme.colors.text.weak}
+              otherProperties={{ marginBottom: metricsSizes["m"] + "px" }}>
+              {intl.formatMessage({
+                defaultMessage: "Re:earth supports Google Sheets, and CSV file to upload.",
+              })}
+            </Text>
+          </Flex>
+          <Flex justify="center">
+            {isLoading ? (
+              <Loading />
+            ) : (
+              <Button
+                large
+                text="Connect your google account"
+                buttonType="primary"
+                onClick={() => handleAuthClick()}
+              />
+            )}
+          </Flex>
+        </>
+      )}
+    </>
+  );
+};
+
+const GdriveIcon = styled(Icon)`
+  margin-right: ${metricsSizes["m"]}px;
+`;
+
+const StyledIcon = styled(Icon)`
+  cursor: pointer;
+`;
+
+const AssetList = styled(Flex)`
+  flex-direction: column;
```

Comment:
Just to keep things clean if you can move this into the component's props with the other flex related attributes that will keep things easier to change/troubleshoot later